runtime/src/asset.rs

```diff
@@ -973,58 +973,77 @@ decl_module! {
+            ensure!(Self::is_owner(&ticker, did), "caller is not the owner of this asset");
+
+            let ticker_did = <identity::Module<T>>::get_token_did(&ticker)?;
+            let signer = Signer::from(ticker_did);
+            documents.into_iter().for_each(|doc| {
+                <identity::Module<T>>::add_link(signer, LinkData::Document(doc), None)
+            });
+
             Ok(())
         }
 
-        /// Used to set the details of the document, Only be called by the token owner
+        /// Remove documents for a given token. To be called only by the token owner
         ///
         /// # Arguments
         /// * `origin` Signing key of the token owner
         /// * `did` DID of the token owner
         /// * `ticker` Ticker of the token
-        /// * `name` Name of the document
-        /// * `uri` Off chain URL of the document
-        /// * `document_hash` Hash of the document to proof the incorruptibility of the document
-        pub fn set_document(origin, did: IdentityId, ticker: Ticker, name: Vec<u8>, uri: Vec<u8>, document_hash: Vec<u8>) -> DispatchResult {
+        /// * `doc_ids` Documents to be removed from `ticker`
+        pub fn remove_documents(origin, did: IdentityId, ticker: Ticker, doc_ids: Vec<u64>) -> DispatchResult {
             let sender = ensure_signed(origin)?;
             let sender_signer = Signer::AccountKey(AccountKey::try_from(sender.encode())?);
 
             // Check that sender is allowed to act on behalf of `did`
             ensure!(<identity::Module<T>>::is_signer_authorized(did, &sender_signer), "sender must be a signing key for DID");
             ticker.canonize();
-            ensure!(Self::is_owner(&ticker, did), "user is not authorized");
+            ensure!(Self::is_owner(&ticker, did), "caller is not the owner of this asset");
+
+            let ticker_did = <identity::Module<T>>::get_token_did(&ticker)?;
+            let signer = Signer::from(ticker_did);
+            doc_ids.into_iter().for_each(|doc_id| {
+                <identity::Module<T>>::remove_link(signer, doc_id)
+            });
 
-            <Documents<T>>::insert((ticker, name), (uri, document_hash, <pallet_timestamp::Module<T>>::get()));
             Ok(())
         }
 
-        /// Used to remove the document details for the given token, Only be called by the token owner
+        /// Update document details for the given token, Only be called by the token owner
         ///
         /// # Arguments
         /// * `origin` Signing key of the token owner
         /// * `did` DID of the token owner
         /// * `ticker` Ticker of the token
-        /// * `name` Name of the document
-        pub fn remove_document(origin, did: IdentityId, ticker: Ticker, name: Vec<u8>) -> DispatchResult {
+        /// * `doc_id` Document to be updated
+        /// * `doc` Contents of new document
+        pub fn update_document(origin, did: IdentityId, ticker: Ticker, doc_id: u64, doc: Document) -> DispatchResult {
```

Comment:
Do you think it would be possible to allow a vec<u64> and vec<Document> and update in a batch?

Reply:
For sure, updated the function to take a tuple of doc ids and docs. Multiple docs are updated in a single tx.